pkg/cmd/server/api/v1/types.go

```diff
@@ -138,6 +145,11 @@ type MasterConfig struct {
 
 	// NetworkConfig to be passed to the compiled in network plugin
 	NetworkConfig NetworkConfig `json:"networkConfig"`
+
+	// Enable or disable OpenShift components.
+	// TODO: find out how to write this field only if any of its field contains
+	// non-default value.
+	FeatureConfig FeatureConfig `json:"-"`
```

Comment:
To expand on "separate fields".  I think that `DisabledFeatures` are distinct from `EnabledFeatures`.  I see `EnabledFeatures` more as `ExperimentalFeatures`.  Things that early adopters may want to turn on while we continue development of them in master.  

I see `DisabledFeatures` as features that our product normally has, but that someone wants to turn off for some reason.